web/src/components/njwds-extended/FileInput.tsx

```diff
@@ -0,0 +1,144 @@
+import { AcceptedFileType, InputFile } from "@businessnjgovnavigator/shared";
+import { ReactElement, useEffect, useLayoutEffect, useRef, useState } from "react";
+
+interface Props {
+  accept?: AcceptedFileType[];
+  className?: string;
+  errorMessage?: string;
+  hasError?: boolean;
+  helperText: string;
+  maxSizeInMegabytes?: number; // filesize in bytes (1,048,576 = 1 MB)
+  value: InputFile | undefined;
+  onChange: (file: InputFile) => void;
+}
+
+const FileTypeToInputString: Record<AcceptedFileType, string> = {
+  PDF: "application/pdf",
+  PNG: "image/png",
+};
+
+const InputStringToFileType: Record<string, AcceptedFileType> = (
+  Object.keys(FileTypeToInputString) as AcceptedFileType[]
+).reduce((acc, key) => {
+  const value = FileTypeToInputString[key];
+  acc[value] = key;
+  return acc;
+}, {} as Record<string, AcceptedFileType>);
```

Comment:
Might be nice to extract this into a generic "flip dictionary" helper method

Reply:
included